libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -494,6 +465,48 @@ impl NodeAPI for Greenlight {
         Ok(client.withdraw(request).await?.into_inner())
     }
 
+    async fn prepare_withdraw(
+        &self,
+        prepare_withdraw_request: PrepareWithdrawRequest,
+    ) -> Result<PrepareWithdrawResponse> {
+        let funds = list_funds(self).await?;
+        let amount = self.on_chain_balance(Some(funds.clone())).await?;
```

Comment:
@ademar111190  It is better to calculate the amount out of the utxos array. It will minimize an option for a race between the method calls.

Reply:
@roeierez sorry I haven't noticed this comment before.
I'm avoiding code duplication here. Is there a way to implement your recommendation without duplicating code?